tcms/testcases/views.py

```diff
@@ -830,15 +827,31 @@ def clone(request, template_name='testcases/clone.html'):
                                  messages.SUCCESS,
                                  _('TestCase cloning was successful'))
             return HttpResponseRedirect(reverse('plans-search'))
-    else:
+
+    def get(self, request):
+        if not self.is_request_data_valid(request):
+            return HttpResponseRedirect(request.META.get('HTTP_REFERER', '/'))
+
         selected_cases = get_selected_testcases(request)
         # Initial the clone case form
         clone_form = CloneCaseForm(initial={
             'case': selected_cases,
         })
         clone_form.populate(case_ids=selected_cases)
 
-    context = {
-        'form': clone_form,
-    }
-    return render(request, template_name, context)
+        context = {
+            'form': clone_form,
+        }
+        return render(request, self.template_name, context)
+
+    def is_request_data_valid(self, request):
```

Comment:
since this is a private method and not going to be used from the outside rename to `_is_request_data_valid`